client/deck-validator.js

```diff
@@ -298,19 +325,20 @@ class DeckValidator {
         }
 
         let allAgendas = [deck.agenda].concat(deck.bannerCards || []);
-        return _.compact(allAgendas.map(agenda => agendaRules[agenda.code]));
+        return allAgendas.map(agenda => agendaRules[agenda.code]).filter(a => !!a);
     }
 
     combineValidationRules(validators) {
-        let mayIncludeFuncs = _.compact(_.pluck(validators, 'mayInclude'));
-        let cannotIncludeFuncs = _.compact(_.pluck(validators, 'cannotInclude'));
-        let combinedRules = _.reduce(validators, (rules, validator) => rules.concat(validator.rules || []), []);
+        let mayIncludeFuncs = validators.map(validator => validator.mayInclude).filter(v => !!v);
+        let cannotIncludeFuncs = validators.map(validator => validator.cannotInclude).filter(v => !!v);
+        let combinedRules = validators.reduce((rules, validator) => rules.concat(validator.rules || []), []);
         let combined = {
-            mayInclude: card => _.any(mayIncludeFuncs, func => func(card)),
-            cannotInclude: card => _.any(cannotIncludeFuncs, func => func(card)),
+            mayInclude: card => mayIncludeFuncs.every(func => func(card)),
```

Comment:
This raises red flags with me as I've changed the logic and I don't think I should have had to, would appreciate extra scrutiny here

Reply:
What situation led you to change it? I believe the previous logic should have been correct. This will fail on banner decks - the faction rules `mayInclude` requires a card be the faction or neutral. The banner `mayInclude` requires a card be the banner faction and non-loyal. It can't ever be both unless you do something weird like Lannister bannered w/ Lannister.

Reply:
Putting a non banner non faction card in the deck wouldn't trigger a validation error until I changed it.  I'm really not sure why..